src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/JSImportExportTest.cs

```diff
@@ -31,6 +32,15 @@ public async Task MultipleImportAsync()
             Assert.Same(instance1, instance2);
         }
 
+        [Fact]
+        public async Task CancelableImportAsync()
+        {
+            var cts = new CancellationTokenSource();
+            cts.Cancel();            
+            var actualEx = await Assert.ThrowsAsync<JSException>(async () => await JSHost.ImportAsync("JavaScriptTestHelper", "./JavaScriptTestHelper.mjs", cts.Token));
```

Comment:
* Is there a way to reliably test that the operation gets cancelled in the middle? Passing an already cancelled token is a trivial case that is optimized in other asynchroonous functions. (and you can use `new CancellationToken(true)` instead of creating a CTS)
* Shouldn't it throw `OperationCancelledException`?

Reply:
The underlying JS dynamic `import()` is not cancelable.
The JS promises are rather "abandoned" than really canceled. We only reject the JS promise.
(In general, we don't know what else registered continuation on that promise on JS side)

That creates JavaScript error, not managed error and we marshal it as `JSException` consistently with all other JS Errors.

`fetch` is bit better because it has `AbortController`.

Reply:
@pavelsavara Did you tried something like this?

```
var cts = new CancellationTokenSource();
var exTask = Assert.ThrowsAsync<JSException>(async () => await JSHost.ImportAsync("JavaScriptTestHelper", "./JavaScriptTestHelper.mjs", cts.Token));

cts.Cancel();

var actualEx = await exTask;
```

Reply:
added to the unit test, it works fine